### PR TITLE
Implement cov function in keras.ops

### LINKED ISSUE #23493: Add keras.ops.cov
## Summary

`keras.ops.corrcoef` is available, but the covariance matrix it normalizes is not. NumPy, JAX and PyTorch all ship `cov` alongside `corrcoef`.

## Motivation

Covariance is the base quantity and `corrcoef` is derived from it, so having the derived one without the base is an odd place to stop. It also comes up on its own in whitening, Mahalanobis distance, Gaussian log likelihoods and PCA, none of which want the correlation form.

Without it the centered matmul has to be written by hand, and then repeated per backend if the code needs to stay backend agnostic.

## Proposed API

```python
keras.ops.cov(x)
```

Matching `np.cov` for a 2D input of shape `(N, D)`, where N is the number of variables and D the number of observations, normalized by `D - 1`. A 1D input, or a single variable, yields a scalar.

---

Yes, I would like to send a PR.

### PR DESCRIPTION
**Pull request by Neilblaze**
## Description

Adds `keras.ops.cov`, which estimates the covariance matrix of the variables in a 2D input of shape `(N, D)`, normalized by `D - 1`. Matches `np.cov`, including the scalar result for a 1D input or a single variable.

Supported across the NumPy, TensorFlow, PyTorch, JAX and OpenVINO backends.

There is also a backend-agnostic fallback, following the shape `cbrt`, `hsplit` and `vsplit` use, so a backend without its own `cov` still gets the op. The correctness and dtype tests run under both settings of the flag, so that path is covered too.

Fixes #23493 

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -10162,3 +10162,46 @@ def column_stack(xs):
...
+    Args:
+        x: A 2D tensor of shape `(N, D)`, where N is the number of variables
+           and D is the number of observations.
+
+    Returns:
+        A tensor of shape `(N, N)` representing the covariance matrix.
+    """
+    if any_symbolic_tensors((x,)):
+        return Cov().symbolic_call(x)
+    return backend.numpy.cov(x)
```
Comment (hertschuh) [inline keras/src/ops/numpy.py]: Can you add a fallback implementation for other backends?

Here are some examples to show the pattern:
- https://github.com/keras-team/keras/pull/23570
- https://github.com/keras-team/keras/pull/23541

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for adding this!

File: keras/src/ops/numpy.py (hunk 1 above)
Comment (Neilblaze) [reply keras/src/ops/numpy.py]: @hertschuh Addressed. PTAL!

**Review by hertschuh — APPROVED** [review]
Thanks!
